build_tools/pkgci/bisect/bisect_packages.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python3
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Dev package bisect script.
+
+This connects the `git bisect` tool (https://git-scm.com/docs/git-bisect)
+with IREE's package builds, allowing developers to run tests through commit
+history efficiently. For example, this can be used to spot at which commit
+an `iree-compile` command started failing.
+
+Requirements:
+    git     (https://git-scm.com/)
+    gh      (https://cli.github.com/)
+    Linux   (at least until IREE builds packages for other systems at each commit)
+    Python 3.11
```

Comment:
Do we want to loosen python requirements?

Reply:
pkgci currently only builds packages for Python 3.11: https://github.com/iree-org/iree/blob/8205885e4922c401714032f591d4cf36fbec3178/build_tools/pkgci/build_linux_packages.sh#L68

If we spent the CI time to build for other Python versions then this could be relaxed. I think we may be able to share the build cache between python versions or adjust the package builds in some other way to amortize the costs.

Reply:
Ahh I see okay